02-tamagotchi-interaction/src/lib.rs

```diff
@@ -28,17 +48,72 @@ extern "C" fn handle() {
     };
     match action {
         TmgAction::Name => {
-            debug!("TmgAction::Name {:?}", tamagotchi.name);
             msg::reply(TmgEvent::Name(tamagotchi.name.clone()), 0)
                 .expect("Error in sending reply TmgEvent::Name");
         }
         TmgAction::Age => {
             let age = exec::block_timestamp() - tamagotchi.date_of_birth;
-            debug!("TmgAction::Age {:?}", age);
             msg::reply(TmgEvent::Age(age), 0).expect("Error in sending reply TmgEvent::Age");
         }
+        TmgAction::Feed => {
+            // To determine the number of blocks when the Tamagotchi last ate
+            let current_block: u64 = exec::block_height().into();
+
+            // we get to update the fed and fed_block
+
+            // we use saturating_sub so that the subtraction result won't be below 0
+            let fed = tamagotchi
+                .fed
+                .saturating_sub((current_block - tamagotchi.fed_block) * HUNGER_PER_BLOCK);
```

Comment:
As I see, your Tamagotchi becomes more hungry even after the `TmgAction::Feed` action. It would be more accurate to calculate life factors on each command.

Reply:
Hi @shamilsan would you mind giving me a hint on how to determine if a tamagotchi is getting more hungry? I thought the lower the `fed` value (closer to 1), the hungrier it is

Reply:
The calculation is correct; I mean that we are to update the `fed`, `entertained`, and `slept` fields in every `handle`, not only in related branches. Now, you decrease `fed` only when feeding, `entertained` when entertaining, etc.